frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/services/MfaResetIPVAuthorizationService.java

```diff
@@ -0,0 +1,200 @@
+    private final AuditService auditService;
+    private final CloudwatchMetricsService cloudwatchMetricsService;
+
+    public MfaResetIPVAuthorizationService(
+            ConfigurationService configurationService,
+            JwtService jwtService,
+            TokenService tokenService,
+            RedisConnectionService redisConnectionService,
+            AuditService auditService,
+            CloudwatchMetricsService cloudwatchMetricsService) {
+        this(
+                configurationService,
+                new NowClock(Clock.systemUTC()),
+                jwtService,
+                tokenService,
+                redisConnectionService,
+                auditService,
+                cloudwatchMetricsService);
+    }
+
+    public MfaResetIPVAuthorizationService(
+            ConfigurationService configurationService,
+            NowClock nowClock,
+            JwtService jwtService,
+            TokenService tokenService,
+            RedisConnectionService redisConnectionService,
+            AuditService auditService,
+            CloudwatchMetricsService cloudwatchMetricsService) {
+        this.configurationService = configurationService;
+        this.nowClock = nowClock;
+        this.jwtService = jwtService;
+        this.tokenService = tokenService;
+        this.redisConnectionService = redisConnectionService;
+        this.auditService = auditService;
+        this.cloudwatchMetricsService = cloudwatchMetricsService;
+    }
+
+    public APIGatewayProxyResponseEvent sendMfaResetRequestToIPV(
+            Subject subject, String clientSessionId, Session session, AuditContext auditContext) {
+
+        State state = new State();
+        ClaimsSetRequest claims = buildMfaResetClaimsRequest(subject);
+        EncryptedJWT requestJWT =
+                constructMfaResetAuthorizationJWT(state, subject, claims, clientSessionId);
+
+        AuthorizationRequest.Builder authRequestBuilder =
+                new AuthorizationRequest.Builder(
+                                new ResponseType(ResponseType.Value.CODE),
+                                new ClientID(configurationService.getIPVAuthorisationClientId()))
+                        .endpointURI(configurationService.getIPVAuthorisationURI())
+                        .requestObject(requestJWT);
+
+        AuthorizationRequest ipvAuthorisationRequest = authRequestBuilder.build();
+        String ipvAuthorisationRequestURI = ipvAuthorisationRequest.toURI().toString();
+        LOG.info("MFA reset JAR created, redirect URI {}", ipvAuthorisationRequestURI);
+
+        storeState(session.getSessionId(), state);
+
+        auditService.submitAuditEvent(REVERIFY_AUTHORISATION_REQUESTED, auditContext);
+
+        cloudwatchMetricsService.incrementMfaResetHandoffCount();
+
+        return generateApiGatewayProxyResponse(
+                302, "", Map.of(ResponseHeaders.LOCATION, ipvAuthorisationRequestURI), null);
```

Comment:
This wants to be a 200, and we need to return a payload like `{"authorize_url":...}` or something. This isn't a request from the user's browser here, it's a request from the Node Express frontend which will then serve the user the 302.